lxd/main_forkusernsexec.go

```diff
@@ -221,6 +222,10 @@ out_reap:
 	if (!argvp || !*argvp)
 		return log_error(EXIT_FAILURE, "No command specified");
 
+	ret = lxd_close_range(FD_PIPE_GIDMAP + 1, UINT_MAX, CLOSE_RANGE_CLOEXEC);
```

Comment:
Should this be FD_PIPE_UIDMAP+1?

Also, I am going to need to pass additional FDs into this command, in addition to 3 and 4, so just want to check the additional ones won't get closed?

Reply:
> Should this be FD_PIPE_UIDMAP+1?

It needs to be FD_PIPE_GIDMAP and +1 as the fd passed as the first argument will be closed as well.

> 
> Also, I am going to need to pass additional FDs into this command, in addition to 3 and 4, so just want to check the additional ones won't get closed?

You need the program you're execing to inherit fds? If that's the case we might just not do this check at all.

Reply:
Yes I will need to pass, at least, 1 other file handle in additional to FD_PIPE_UIDMAP and FD_PIPE_GIDMAP

Reply:
See https://github.com/lxc/lxd/blob/aca7736aca79b882fdd43bd02eaf4053b2a17676/lxd/device/device_utils_disk.go#L298 where I will wrap that call with the forkusernsexec command, and then pass fd 5 as the socket, changing the `--fd 3` arg to `--fd 5`.

Reply:
It works right now btw :)